0015-authority.md

```diff
@@ -0,0 +1,478 @@
+
+## Modularity
+Currently when developing with MLAPI it is very hard to develop modular components. Each component might have different logic and a list of expectations which are not compatible with each other. While the authority system will not fully solve this problem it will help developers which create components which can be combined with other components using the same authority. This allows a user to build reusable components which is a core feature of Unity when creating single player games.
+
+## Cross/Multi Platform
+Being able to launch a multiplayer game on multiple platforms is important for a lot of our users. Different platforms have very different networking needs:
+- Mobile : high latency, high jitter, almost no hackers
+- PC: low latency, low jitter, many hackers
+- Console: low latency, low jitter, almost no hackers
+
+Based on those platforms conditions different authority modes might be more ideal for some of them. `Mobile` for instance could profit from `Owner Authority` while `PC` or `Console` might want server authority.
+
+The authority feature will allow to develop a game once and then support all those modes.
+
+## Unify Existing Components
+`NetworkTransform` has already an authority switch. NetworkNavMeshAgent and NetworkAnimator will need one too. Instead of solving the same problem over and over again we will solve it once.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+NetworkObject have an `Authority` switch. The Authority switch can be used when creating a `NetworkPrefab` to decide which authority model the `NetworkObject` should use.
+
+Currently there are two `Authority` modes:
+- Owner: The owner of the `NetworkObject` has authority over it.
+- Server: The server always has authority over the `NetworkObject.
+
+![dropdown NetworkObject](./0015-authority/dropdown-ownership-networkobject.png)
+
+We might add new authority modes in the future.
+
+Having Authority over a `NetworkObject` does the following
+- There is a `HasAuthority` property on the `NetworkObject` and each `NetworkBehaviour` which will be set accordingly.
+- Built in MLAPI components such as `NetworkTransform` or `NetworkAnimator` will synchronize the state of the at
+- NetworkVariables with `ReadPermission = NetworkVariablePermission.Authority` can be read from. (write permission also exists)
+
+Having Authority over a `NetworkObject` **does not**: change MLAPI features such as how NetworkVariables or RPCs are sent/received etc. In the future we might have authority modes such as `Predicted` which will change this. But this is not part of the authority mode but will be implemented by the specific feature like `prediction`.
```

Comment:
"predicted" would still remain server authoritative in that case. I'm not sure why the authority switch would be aware of that. Do you mean that a server authoritative netvar would prevent updating that netvar client side? I would keep it like we have right now, where the server throws errors when receiving updates from netvars it shouldn't.
This way client side you can still allow changing a server authoritative netvar, but don't sent the value change if you're predicted.

Reply:
> This way client side you can still allow changing a server authoritative netvar, but don't sent the value change if you're predicted.

This is was what I was trying to say. If we do prediction one way of doing this would be as you mentioned to allow the client to write but then not synchronize those changes.